compio-driver/src/iocp/op.rs

```diff
@@ -906,18 +907,22 @@ impl<T: IoVectoredBuf, C: IoBuf, S: AsRawFd> OpCode for SendMsg<T, C, S> {
     unsafe fn operate(self: Pin<&mut Self>, optr: *mut OVERLAPPED) -> Poll<io::Result<usize>> {
         let this = self.get_unchecked_mut();
 
-        let slices = this.buffer.io_slices();
-        let msg = WSAMSG {
-            name: this.addr.as_ptr() as _,
-            namelen: this.addr.len(),
-            lpBuffers: slices.as_ptr() as _,
-            dwBufferCount: slices.len() as _,
-            Control: std::mem::transmute::<IoSlice, WSABUF>(this.control.as_io_slice()),
-            dwFlags: 0,
-        };
+        this.slices = this.buffer.io_slices();
+        this.msg.name = this.addr.as_ptr() as _;
+        this.msg.namelen = this.addr.len();
+        this.msg.lpBuffers = this.slices.as_ptr() as _;
+        this.msg.dwBufferCount = this.slices.len() as _;
+        this.msg.Control = std::mem::transmute::<IoSlice, WSABUF>(this.control.as_io_slice());
 
         let mut sent = 0;
-        let res = WSASendMsg(this.fd.as_raw_fd() as _, &msg, 0, &mut sent, optr, None);
+        let res = WSASendMsg(
```

Comment:
According to the documents, `WSASendMsg` should be retrieved from `WSAIoctl`, just like `WSARecvMsg`. Don't call it directly.

Reply:
But the fact is that everyone is happy to use it, for example in [socket2](https://github.com/rust-lang/socket2/blob/3a938932829ea6ee3025d2d7a86c7b095c76e6c3/src/sys/windows.rs#L675). I believe it's not necessary to strictly follow the document since `windows-sys` crate exports this function.